### PR TITLE
ci: add nightly performance testing, update localnet Docker image to Debian Bookworm, remove unnecessary build-jet runners.

### PR DESCRIPTION
**Pull request by gzukel**
# Description

ci: add nightly performance testing, update localnet Docker image to Debian Bookworm, remove unnecessary build-jet runners.

Closes: DEVOP-699

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] CI Updates

# How Has This Been Tested?

The docker image has been built locally to validate it builds and tested. Need to merge to register the performance pipeline for testing.

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 
- [x] Locally

# Checklist:

- [x] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -202,7 +202,7 @@ jobs:
   build-test:
     needs:
       - check_branch
-    runs-on: buildjet-4vcpu-ubuntu-2204
+    runs-on: ubuntu-latest
```
Comment (gartnera) [inline .github/workflows/publish-release.yml:205]:
```suggestion
    runs-on: ubuntu-22.04
```
I would prefer to explicitly select a version.

File: .github/workflows/publish-release.yml [hunk 2]
```diff
@@ -411,7 +411,7 @@ jobs:
       - e2e-admin-tests
       - e2e-upgrade-test
       - check_branch
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```
Comment (gartnera) [inline .github/workflows/publish-release.yml:414]: maybe we should upgrade `ghcr.io/goreleaser/goreleaser-cross` while we're at it? Without upgrading this we don't actually increase the minimum glibc version.

**Comment by gartnera** [comment]
```
zetae2e: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by zetae2e)
```
e2e has glibc version missmatch issues. You need to upgrade the orchestrator too.

File: changelog.md [hunk 3]
```diff
@@ -54,7 +54,12 @@
 * [2256](https://github.com/zeta-chain/node/pull/2256) - fix rate limiter falsely included reverted non-withdraw cctxs
 
 ### CI
-* [2268](https://github.com/zeta-chain/node/pull/2268) - CI: updated the publish-release pipeline to utilize the Github Actions Ubuntu 20.04 Runners.
+
+* [2285](https://github.com/zeta-chain/node/pull/2285) - added nightly EVM performance testing pipeline.
+* [2285](https://github.com/zeta-chain/node/pull/2285) - modified localnet testing docker image to utilitze debian:bookworm.
+* [2285](https://github.com/zeta-chain/node/pull/2285) - removed build-jet runners where applicable.
+* [2285](https://github.com/zeta-chain/node/pull/2285) - removed deprecated/removed upgrade path testing pipeline.
```
Comment (kingpinXD) [inline changelog.md:61]: There should be only one entry per pr, maybe condense the task list into a sentence?

File: .github/workflows/ci-nightly-performance-testing.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      - uses: actions/checkout@v4
+
+      - name: "START:LOCAL:NET"
+        run: | 
+          make start-e2e-test
+
+      - name: "EXECUTE:LOADTESTS"
+        uses: artilleryio/action-cli@v1
+        with:
+          command: run .github/actions/performance-tests/art.yaml --output ./report.json
```
Comment (kingpinXD) [inline .github/workflows/ci-nightly-performance-testing.yaml:22]: From my understanding, does the LoadTest only run queries against a node?
Or are there txs submitted as well?

**Review by lumtis — APPROVED** [review]

File: .github/workflows/ci-nightly-performance-testing.yaml [hunk 5]
```diff
@@ -0,0 +1,34 @@
+name: "NIGHTLY:EVM:PERFORMANCE:TESTING"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *' # Runs every day at midnight
```
Comment (CryptoFewka) [inline .github/workflows/ci-nightly-performance-testing.yaml:6]: Nitpick: Perhaps add the time zone for clarity, as this runs every day at midnight UTC, which is 5 PM Pacific.

File: .github/workflows/ci-nightly-performance-testing.yaml (hunk 4 above)
Comment (kingpinXD) [reply .github/workflows/ci-nightly-performance-testing.yaml:22]: Okay , I'm just curious: what is the motivation behind these tests? As queries will only test the node serving the request

File: .github/workflows/ci-nightly-performance-testing.yaml [hunk 6]
```diff
@@ -0,0 +1,34 @@
+name: "NIGHTLY:EVM:PERFORMANCE:TESTING"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 0 * * *' # Runs every day at midnight UTC
+
+jobs:
+  nightly_evm_api_performance_test:
+    name: "NIGHTLY:EMV:API:PERFORMANCE:TEST"
```
Comment (gartnera) [inline .github/workflows/ci-nightly-performance-testing.yaml:10]:
```suggestion
    name: "NIGHTLY:EVM:API:PERFORMANCE:TEST"
```

**Review by gartnera — APPROVED** [review]

**Pull request merged**